### PR TITLE
Add timeout out in dsl

### PR DESCRIPTION
**Pull request by gaoning777**
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/1465)

### CODE REVIEW

File: sdk/python/kfp/compiler/_op_to_template.py [hunk 1]
```diff
@@ -245,6 +245,10 @@ def _op_to_template(op: BaseOp):
     if processed_op.num_retries:
         template['retryStrategy'] = {'limit': processed_op.num_retries}
 
+    # timeout
+    if processed_op.timeout:
+        template['activeDeadlineSeconds'] = processed_op.timeout
```
Comment (hongye-sun) [inline sdk/python/kfp/compiler/_op_to_template.py]: Could you add one more test to cover the setting?

Comment (gaoning777) [reply sdk/python/kfp/compiler/_op_to_template.py]: Good point. Done.

**Comment by Ark-kun** [comment]
It would be best if the test checked the actual behavior (as opposed to just checking that nothing has changed).

For example you can call `_op_to_template` or `_compile` and check whether the resulting template has `activeDeadlineSeconds` set to the same value.

**Comment by gaoning777** [comment]
I'm confused. Isn't the test checking the activeDeadlineSeconds value?

**Pull request merged**